src/main/java/nl/andrewl/emaildatasetbrowser/view/search/LuceneSearchPanel.java

```diff
@@ -59,6 +62,14 @@ public LuceneSearchPanel(EmailViewPanel emailViewPanel) {
         exportPanel.add(exportButton);
         bottomPanel.add(exportPanel);
 
+        JPanel progressPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+        showProgressButton
+                .setSelected(EmailDatasetBrowser.getPreferences().getBoolean(PREFERENCES_SHOW_PROGRESS, true));
+        showProgressButton.addActionListener((e) -> EmailDatasetBrowser.getPreferences()
+                .putBoolean(PREFERENCES_SHOW_PROGRESS, ((JToggleButton) e.getSource()).isSelected()));
+        progressPanel.add(showProgressButton);
+        bottomPanel.add(progressPanel);
+
```

Comment:
The progress dialog is useful when your query is busted (e.g. when you have too many operants or some syntactical error). I can make it that if everything goes right, it doesn't show, but if an exception is thrown, the dialog pops up? This could be some other dialog too, though.

Reply:
Yeah, it would probably be better if you just show a warning popup.